app/models/theme.rb

```diff
@@ -1024,6 +1028,40 @@ def user_selectable_count
     UserOption.where(theme_ids: [id]).count
   end
 
+  def find_or_create_owned_color_palette
+    Theme.transaction do
+      next self.owned_color_palette if self.owned_color_palette
+
+      palette = self.color_palette || ColorScheme.base
+
+      copy = palette.dup
+      copy.theme_id = self.id
+      copy.base_scheme_id = nil
+      copy.user_selectable = false
+      copy.via_wizard = false
+      copy.save!
+
+      result =
+        ThemeColorScheme.insert_all(
+          [{ theme_id: self.id, color_scheme_id: copy.id }],
+          unique_by: :index_theme_color_schemes_on_theme_id,
+        )
```

Comment:
Just curious, why do we need to use `insert_all` here when we can just use `create!` and let the database handle the uniqueness validation.

Reply:
If `create!` fails due to uniqueness validation (or any other reason), the whole transaction is "poisoned" due to the failed `INSERT` statement, and that will cause all the statements that follow to fail as well. So I need a way to make the `INSERT` statement fail "peacefully" and `insert_all` does the job since it has `ON CONFLICT DO NOTHING`.